### ISSUE TITLE
Refactoring: Introduce Variable(inplace)

### ISSUE THREAD

**Issue by as3boyan** [HaxeFoundation/intellij-haxe#228]
IntelliJ IDEA 13.1.6

```
date1.getDate()
```

Select this text and use Refactoring->Extract->Variable, it seems to work, but also throws this error

```
[ 124455]  ERROR - inplace.VariableInplaceRenamer - COMPONENT_NAME 'date' by com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler$HaxeInplaceVariableIntroducer 
java.lang.Throwable
    at com.intellij.openapi.diagnostic.Logger.error(Logger.java:113)
    at com.intellij.refactoring.rename.inplace.InplaceRefactoring.a(InplaceRefactoring.java:721)
    at com.intellij.refactoring.rename.inplace.InplaceRefactoring.buildTemplateAndStart(InplaceRefactoring.java:273)
    at com.intellij.refactoring.rename.inplace.InplaceRefactoring.performInplaceRefactoring(InplaceRefactoring.java:213)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.performInplaceIntroduce(HaxeIntroduceHandler.java:378)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler$3.pass(HaxeIntroduceHandler.java:267)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler$3.pass(HaxeIntroduceHandler.java:263)
    at com.intellij.refactoring.introduce.inplace.OccurrencesChooser.showChooser(OccurrencesChooser.java:81)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.performActionOnElementOccurrences(HaxeIntroduceHandler.java:260)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.performActionOnElement(HaxeIntroduceHandler.java:249)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.performAction(HaxeIntroduceHandler.java:167)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.invoke(HaxeIntroduceHandler.java:104)
    at com.intellij.refactoring.actions.BaseRefactoringAction.actionPerformed(BaseRefactoringAction.java:122)
    at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:164)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter$1.run(ActionMenuItem.java:266)
    at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:926)
    at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:124)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:236)
    at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1995)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:105)
    at com.intellij.ui.plaf.beg.BegMenuItemUI.a(BegMenuItemUI.java:512)
    at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:44)
    at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:532)
    at java.awt.Component.processMouseEvent(Component.java:6297)
    at javax.swing.JComponent.processMouseEvent(JComponent.java:3275)
    at java.awt.Component.processEvent(Component.java:6062)
    at java.awt.Container.processEvent(Container.java:2039)
    at java.awt.Component.dispatchEventImpl(Component.java:4660)
    at java.awt.Container.dispatchEventImpl(Container.java:2097)
    at java.awt.Component.dispatchEvent(Component.java:4488)
    at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4575)
    at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4236)
    at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4166)
    at java.awt.Container.dispatchEventImpl(Container.java:2083)
    at java.awt.Window.dispatchEventImpl(Window.java:2489)
    at java.awt.Component.dispatchEvent(Component.java:4488)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:674)
    at java.awt.EventQueue.access$400(EventQueue.java:81)
    at java.awt.EventQueue$2.run(EventQueue.java:633)
    at java.awt.EventQueue$2.run(EventQueue.java:631)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:98)
    at java.awt.EventQueue$3.run(EventQueue.java:647)
    at java.awt.EventQueue$3.run(EventQueue.java:645)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:644)
    at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:697)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:520)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:335)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:269)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:184)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:174)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:169)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:161)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
[ 124457]  ERROR - inplace.VariableInplaceRenamer - IntelliJ IDEA 13.1.6  Build #IU-135.1306 
[ 124457]  ERROR - inplace.VariableInplaceRenamer - JDK: 1.6.0_45 
[ 124457]  ERROR - inplace.VariableInplaceRenamer - VM: Java HotSpot(TM) 64-Bit Server VM 
[ 124457]  ERROR - inplace.VariableInplaceRenamer - Vendor: Sun Microsystems Inc. 
[ 124457]  ERROR - inplace.VariableInplaceRenamer - OS: Windows 8 
[ 124457]  ERROR - inplace.VariableInplaceRenamer - Last Action: IntroduceVariable 
[ 124457]  ERROR - llij.ide.plugins.PluginManager - null 
java.lang.NullPointerException
    at com.intellij.refactoring.rename.inplace.InplaceRefactoring.buildTemplateAndStart(InplaceRefactoring.java:287)
    at com.intellij.refactoring.rename.inplace.InplaceRefactoring.performInplaceRefactoring(InplaceRefactoring.java:213)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.performInplaceIntroduce(HaxeIntroduceHandler.java:378)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler$3.pass(HaxeIntroduceHandler.java:267)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler$3.pass(HaxeIntroduceHandler.java:263)
    at com.intellij.refactoring.introduce.inplace.OccurrencesChooser.showChooser(OccurrencesChooser.java:81)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.performActionOnElementOccurrences(HaxeIntroduceHandler.java:260)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.performActionOnElement(HaxeIntroduceHandler.java:249)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.performAction(HaxeIntroduceHandler.java:167)
    at com.intellij.plugins.haxe.ide.refactoring.introduce.HaxeIntroduceHandler.invoke(HaxeIntroduceHandler.java:104)
    at com.intellij.refactoring.actions.BaseRefactoringAction.actionPerformed(BaseRefactoringAction.java:122)
    at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:164)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter$1.run(ActionMenuItem.java:266)
    at com.intellij.openapi.wm.impl.FocusManagerImpl.runOnOwnContext(FocusManagerImpl.java:926)
    at com.intellij.openapi.wm.impl.IdeFocusManagerImpl.runOnOwnContext(IdeFocusManagerImpl.java:124)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem$ActionTransmitter.actionPerformed(ActionMenuItem.java:236)
    at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:1995)
    at com.intellij.openapi.actionSystem.impl.ActionMenuItem.fireActionPerformed(ActionMenuItem.java:105)
    at com.intellij.ui.plaf.beg.BegMenuItemUI.a(BegMenuItemUI.java:512)
    at com.intellij.ui.plaf.beg.BegMenuItemUI.access$300(BegMenuItemUI.java:44)
    at com.intellij.ui.plaf.beg.BegMenuItemUI$MyMouseInputHandler.mouseReleased(BegMenuItemUI.java:532)
    at java.awt.Component.processMouseEvent(Component.java:6297)
    at javax.swing.JComponent.processMouseEvent(JComponent.java:3275)
    at java.awt.Component.processEvent(Component.java:6062)
    at java.awt.Container.processEvent(Container.java:2039)
    at java.awt.Component.dispatchEventImpl(Component.java:4660)
    at java.awt.Container.dispatchEventImpl(Container.java:2097)
    at java.awt.Component.dispatchEvent(Component.java:4488)
    at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4575)
    at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4236)
    at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4166)
    at java.awt.Container.dispatchEventImpl(Container.java:2083)
    at java.awt.Window.dispatchEventImpl(Window.java:2489)
    at java.awt.Component.dispatchEvent(Component.java:4488)
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:674)
    at java.awt.EventQueue.access$400(EventQueue.java:81)
    at java.awt.EventQueue$2.run(EventQueue.java:633)
    at java.awt.EventQueue$2.run(EventQueue.java:631)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:98)
    at java.awt.EventQueue$3.run(EventQueue.java:647)
    at java.awt.EventQueue$3.run(EventQueue.java:645)
    at java.security.AccessController.doPrivileged(Native Method)
    at java.security.AccessControlContext$1.doIntersectionPrivilege(AccessControlContext.java:87)
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:644)
    at com.intellij.ide.IdeEventQueue.e(IdeEventQueue.java:697)
    at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:520)
    at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:335)
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:269)
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:184)
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:174)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:169)
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:161)
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:122)
[ 124461]  ERROR - llij.ide.plugins.PluginManager - IntelliJ IDEA 13.1.6  Build #IU-135.1306 
[ 124461]  ERROR - llij.ide.plugins.PluginManager - JDK: 1.6.0_45 
[ 124461]  ERROR - llij.ide.plugins.PluginManager - VM: Java HotSpot(TM) 64-Bit Server VM 
[ 124461]  ERROR - llij.ide.plugins.PluginManager - Vendor: Sun Microsystems Inc. 
[ 124461]  ERROR - llij.ide.plugins.PluginManager - OS: Windows 8 
[ 124461]  ERROR - llij.ide.plugins.PluginManager - Last Action: IntroduceVariable 
```

**Comment by EricBishton** [comment]
A fix is in PR #902.  A number of issues could cause a null pointer to be passed to buildTemplateAndStart().  They have been addressed.

**Comment by EricBishton** [comment]
Closing, since it was fixed a while back.
